components/scream/src/share/io/output_manager.hpp

```diff
@@ -0,0 +1,271 @@
+    repo_set            = true;
+    m_grids_manager     = gm;
+    gm_set              = true;
+    m_runtype_restart   = runtype_restart;
+  }
+
+  OutputManager( const ekat::Comm& comm, const ekat::ParameterList& params,
+                 const std::shared_ptr<const FieldRepository<Real>>& repo,
+                 const std::shared_ptr<const GridsManager>& gm)
+  {
+    atm_comm            = comm;
+    m_params            = params;
+    param_set           = true;
+    m_device_field_repo = repo;
+    repo_set            = true;
+    m_grids_manager     = gm;
+    gm_set              = true;
+  }
+
+  void init();
+  void run(util::TimeStamp& current_ts);
+  void run(Real current_ts);
+  void finalize();
+
+  void new_output(const ekat::ParameterList& params);
+  void new_output(const ekat::ParameterList& params,const bool runtype_restart);
+  void set_comm(const ekat::Comm& comm) { atm_comm = comm; }
+  void set_params(const ekat::ParameterList& params) { m_params=params; param_set=true;}
+  void set_grids(const std::shared_ptr<const GridsManager>& gm) { m_grids_manager = gm; gm_set=true;}
+  void set_repo(const std::shared_ptr<const FieldRepository<Real>>& repo) { m_device_field_repo = repo; repo_set=true;}
+  void set_runtype_restart(const bool bval) { m_runtype_restart = bval; }
+  void make_restart_param_list(ekat::ParameterList& params);
+
+protected:
+  std::vector<std::shared_ptr<output_type>>    m_output_streams;
+  ekat::Comm                                   atm_comm;
+  ekat::Comm                                   pio_comm; 
+  ekat::ParameterList                          m_params;
+  std::shared_ptr<const FieldRepository<Real>> m_device_field_repo;
+  std::shared_ptr<const GridsManager>          m_grids_manager;
+
+  bool                                 param_set = false;
+  bool                                 gm_set    = false;
+  bool                                 repo_set  = false;
+  bool                        m_runtype_restart  = false;
+
+}; // class OutputManager
+/*===============================================================================================*/
+/* Short function to add a new output stream to the output manager.  By making this a stand-alone
+ * function it is possible to add an output stream after initialization has been called.
+ * Note: new_output requires a parameter list with all of the output control information for this
+ * stream.  See scorpio_output.hpp for more information on what the parameter list needs.         */
+inline void OutputManager::new_output(const ekat::ParameterList& params)
+{
+  auto output_instance = std::make_shared<output_type>(pio_comm,params,m_device_field_repo,m_grids_manager,m_runtype_restart);
+  output_instance->init();
+  m_output_streams.push_back(output_instance);
+}
+/* --------------------------------------------------------------------- */
+inline void OutputManager::new_output(const ekat::ParameterList& params, const bool runtype_restart)
```

Comment:
You could give `runtype_restart` a default value and get rid of the one-argument version of this method.

Reply:
I think I will save this comment for a future PR.  I'm still not sure how `new_output` will be used.  I think that we could probably get rid of the single argument version but I don't want to go down a path that might take longer than expected and delay this PR anymore.